conn/zstrm.c

```diff
@@ -105,9 +105,9 @@ static int zstrm_close(struct Connection *conn)
 
   mutt_debug(LL_DEBUG5, "read %lu->%lu (%.1fx) wrote %lu<-%lu (%.1fx)\n",
              zctx->read.z.total_in, zctx->read.z.total_out,
-             (float) zctx->read.z.total_out / (float) zctx->read.z.total_in,
+             zctx->read.z.total_out / zctx->read.z.total_in,
              zctx->write.z.total_in, zctx->write.z.total_out,
-             (float) zctx->write.z.total_in / (float) zctx->write.z.total_out);
+             zctx->write.z.total_in / zctx->write.z.total_out);
```

Comment:
I get warnings about this now:
```
conn/zstrm.c:106:25: warning: format ‘%f’ expects argument of type ‘double’, but argument 5 has type ‘long unsigned int’
conn/zstrm.c:106:25: warning: format ‘%f’ expects argument of type ‘double’, but argument 8 has type ‘long unsigned int’
```

The fields: `zctx->read.z.total_in` and `zctx->read.z.total_out` are **`unsigned long`**.
(see `struct z_stream` in `zlib.h`)

Reply:
Casting to `(double)` seems to keep the compiler happy:
```c
mutt_debug(LL_DEBUG5, "read %lu->%lu (%.1fx) wrote %lu<-%lu (%.1fx)\n",
           zctx->read.z.total_in, zctx->read.z.total_out,
           (double) zctx->read.z.total_out / (double) zctx->read.z.total_in,
           zctx->write.z.total_in, zctx->write.z.total_out,
           (double) zctx->write.z.total_in / (double) zctx->write.z.total_out);
```

Reply:
yeah makes sense, sorry. I wonder why this didn't show up for me. 🤔

I changed the cast to `double` now. But there's much more important stuff for you to look at than this cosmetic fixes. Feel free to just close this if it doesn't work.

Reply:
I probably have some `-Wformat-???` options enabled

Reply:
I have `-Wformat=2` which should enable all of them. 🤷🏼